src/main/java/org/terasology/dynamicCities/settlements/SettlementEntityManager.java

```diff
@@ -143,14 +142,13 @@ public void update(float delta) {
         }
         Iterable<EntityRef> uncheckedSiteRegions = entityManager.getEntitiesWith(SiteComponent.class);
         for (EntityRef siteRegion : uncheckedSiteRegions) {
-            boolean checkDistance = checkMinDistance(siteRegion);
-            boolean checkBuildArea = checkBuildArea(siteRegion);
-            if (checkDistance && regionEntityManager.checkSidesLoadedNear(siteRegion)
-                    && checkBuildArea) {
+            CheckSiteSuitabilityEvent checkSiteSuitabilityEvent = new CheckSiteSuitabilityEvent();
+            siteRegion.send(checkSiteSuitabilityEvent);
+            if (checkSiteSuitabilityEvent.getResult() == SettlementFilterResult.SUITABLE) {
```

Comment:
You probably want to use a switch statement here (also helps to spot missed cases, etc.).

```java
switch(checkSiteSuitabilityEvent.getResult()) {
    case SettlementFilterResult.SUITABLE:
      ...;
      break;
  case SettlementFilterResult.UNSUITABLE:
  case SettlementFilterResult.UNKOWN:
      ...;
      break;
}